src/main/java/school/hei/haapi/authentication/SecurityConfiguration.java

```diff
@@ -0,0 +1,26 @@
+package school.hei.haapi.authentication;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    private AwsCognitoJwtAuthFilter awsCognitoJwtAuthenticationFilter;
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+
+        http.headers().cacheControl();
+        http.csrf().disable()
+                .authorizeRequests()
+                .antMatchers("/ping").authenticated()
+                .anyRequest().authenticated()
+                .and()
+                .addFilterBefore(awsCognitoJwtAuthenticationFilter, UsernamePasswordAuthenticationFilter.class);
```

Comment:
UsernamePassword? NOT SURE

Reply:
After making some research, I found that UsernamePasswordAuthenticationFilter processes an authentication form submission (username/password), and when I searched for an alternative everyone who is using token especially jwt always use UsernamePassword...

Reply:
Not conviced. There should be some kind of DefaultAuthenticationFilter before which you can put your CognitoFilter. UserNameFilter is really irrelevant here since we don't use any form

Reply:
My bad, the BasicAuthenticationFilter is enough.